fathom/src/core.rs

```diff
@@ -246,6 +248,18 @@ pub enum Prim {
     FormatArray64,
     /// A format which returns the current position in the input stream.
     FormatStreamPos,
+    /// A format that links to another location in the binary data stream,
+    /// relative to a base position and a 8-bit offset.
+    FormatLink8,
+    /// A format that links to another location in the binary data stream,
+    /// relative to a base position and a 16-bit offset.
+    FormatLink16,
+    /// A format that links to another location in the binary data stream,
+    /// relative to a base position and a 32-bit offset.
+    FormatLink32,
+    /// A format that links to another location in the binary data stream,
+    /// relative to a base position and a 64-bit offset.
```

Comment:
Unsigned offsets, presumably.

Reply:
ahh, that's correct!

Reply:
Hopefully avoiding obvious non-termination issues but idk

Reply:
The only way you could get non-termination is by repeatedly loading the same offset with different types, but you would need an infinite type to make that work. You could come close with something like this:
```
struct Nasty (r: Ref) (n: u32) {
    oops : Link32 r 0 (Nasty r (n + 1))
}
```
Perhaps you could make this worse by evaluating Ackermann's function in the type? :laughing: